v0.41.0 changelog: decomps (#7222)
PennyLaneAI/pennylane: doc/releases/changelog-0.41.0.md

# Release 0.41.0 (current release)

<h4>Resource-efficient Decompositions 🔎</h4>

> A new, experimental graph-based decomposition system is now available in PennyLane, offering a more resource-efficiency and versatility.

Author: ```suggestion
A new, experimental graph-based decomposition system is now available in PennyLane, offering more resource-efficiency and versatility.
```

# Release 0.41.0 (current release)

<h4>Resource-efficient Decompositions 🔎</h4>

> A new, experimental graph-based decomposition system is now available in PennyLane under the `qml.decomposition` 
> module. 

Author: ```suggestion
A new, experimental graph-based decomposition system is now available in PennyLane, offering a more resource-efficiency and versatility.
```

[(#6950)](https://github.com/PennyLaneAI/pennylane/pull/6950)
[(#6951)](https://github.com/PennyLaneAI/pennylane/pull/6951)
[(#6952)](https://github.com/PennyLaneAI/pennylane/pull/6952)
[(#6966)](https://github.com/PennyLaneAI/pennylane/pull/6966)
[(#7045)](https://github.com/PennyLaneAI/pennylane/pull/7045)
[(#7058)](https://github.com/PennyLaneAI/pennylane/pull/7058)
[(#7064)](https://github.com/PennyLaneAI/pennylane/pull/7064)
[(#7149)](https://github.com/PennyLaneAI/pennylane/pull/7149)
[(#7184)](https://github.com/PennyLaneAI/pennylane/pull/7184)

PennyLane's new experimental graph decomposition system offers a resource-efficient and versatile alternative 
to the current system. This is done by traversing an internal graph structure that is weighted by the 
resources (e.g., gate counts) required to decompose down to a given set of gates. 

> This new system is experimental and is disabled by default, but it can be enabled by adding 

Author: ```suggestion
The graph-based system is experimental and is disabled by default, but it can be enabled by adding 
```

  1: ──RX(0.25)─╰Z──RX(-0.25)─╰Z─┤
  ```

>   When an operator within a circuit needs to be decomposed (e.g., when 
>   :func:`qml.transforms.decompose <pennylane.transforms.decompose>` is present), the chosen decomposition 
>   rule is that which leads to the most resource efficient set of gates (e.g., the least amount of gates 
>   produced).

Reviewer: ```suggestion
  When an operator within a circuit needs to be decomposed (e.g., when 
  :func:`qml.transforms.decompose <pennylane.transforms.decompose>` is present), the chosen
  decomposition rule is that which is most resource efficient: resulting in the fewest gates.
```

Author: ```suggestion
  When an operator within a circuit needs to be decomposed (e.g., when 
  :func:`qml.transforms.decompose <pennylane.transforms.decompose>` is present), the chosen
decomposition rule is that which is most resource efficient (results in the smallest gate count).
```

Adaptation of your suggestion 👀

      qml.MultiRZ(theta, wires=wires[:3])
  ```

  More information for defining complex decomposition rules can be found in the documentation for 
  :func:`qml.register_resources <pennylane.register_resources>`.

* The :func:`qml.transforms.decompose <pennylane.transforms.decompose>` transform works when the new 
  decompositions system is enabled and offers the ability to inject new decomposition rules for operators
  in QNodes.
  [(#6966)](https://github.com/PennyLaneAI/pennylane/pull/6966)

>   The with graph-based system enabled, the :func:`qml.transforms.decompose <pennylane.transforms.decompose>` 
>   transform offers the ability to inject new decomposition rules via two new keyword arguments:

Reviewer: ```suggestion
  With the graph-based system enabled, the :func:`qml.transforms.decompose <pennylane.transforms.decompose>` 
  transform offers the ability to inject new decomposition rules via two new keyword arguments:
```

Author: 🥴

>   * `fixed_decomps`: decomposition rules provided to this keyword argument are guaranteed to be used 

Author: ```suggestion
  * `fixed_decomps`: decomposition rules provided to this keyword argument will be used 
```
